node/node.go

```diff
@@ -212,7 +213,44 @@ type Node struct {
 // Assumes [n.CPUTracker] and [n.CPUTargeter] have been initialized.
 func (n *Node) initNetworking(primaryNetVdrs validators.Set) error {
 	currentIPPort := n.Config.IPPort.IPPort()
-	listener, err := net.Listen(constants.NetworkType, fmt.Sprintf(":%d", currentIPPort.Port))
+
+	// Default to binding to all interfaces on the specified port
+	networkType := constants.NetworkType
+	listenAddress := fmt.Sprintf(":%d", currentIPPort.Port)
+
+	// On MacOS, if either loopback address is specified, bind only to that address to minimize
+	// developer friction.
+	//
+	// - MacOS requires a manually-approved firewall exception [1] for each version of a given binary
+	// that wants to bind to all interfaces (i.e. with an address of `:[port]`). Each compiled
+	// version of avalanchego requires a separate exception to be allowed to bind to all interfaces.
+	//
+	// - A firewall exception is not required to bind to a loopback interface, but the only way for
+	// Listen() to bind to loopback for both ipv4 and ipv6 is to bind to all interfaces [2] which
+	// requires an exception.
+	//
+	// - Thus, the only way to start a local avalanche network on MacOS without approving one or more
+	// firewall exceptions for the targeted avalanchego binaries is to bind to loopback for only a
+	// single network type by specifying either `[::1]` (ipv6) or `127.0.0.1` (ipv4).
+	//
+	// 1: https://apple.stackexchange.com/questions/393715/do-you-want-the-application-main-to-accept-incoming-network-connections-pop
+	// 2: https://github.com/golang/go/issues/56998
+	//
+	// TODO(marun) Should this be configurable and/or test-only?
+	//
+	if goruntime.GOOS == "darwin" && currentIPPort.IP.IsLoopback() {
```

Comment:
1. I don't know if there's a specific reason ANR uses ipv6 over ipv4, but this is more of an aside
2. I think we can keep this behavior as default when running this on localhost, since the only time we should be doing this is when testing locally so i think this is not that invasive